### PR TITLE
Remove spec saphostctrl_listinstances

### PR DESCRIPTION
**Pull request by xiangce**
fix： #2610 
Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### CODE REVIEW

**Comment by xiangce** [comment]
@bfahr - This spec is not added to the uploader.json ever. So I didn't create an INSPEC card for this PR. Please have a look.

**Comment by bfahr** [comment]
@xiangce these three datasources look more like parsers than data sources.  Would it be possible to just use the `SAPHostCtrlInstances` parser instead of these 3 datasources?

**Comment by xiangce** [comment]
> @xiangce these three datasources look more like parsers than data sources. Would it be possible to just use the `SAPHostCtrlInstances` parser instead of these 3 datasources?

@bfahr - currently, there are 2 `foreach_execute` specs depend on the result got from the the `SAPHostCtrlInstances` parser
```python
sap_hdb_version = foreach_execute(sap_sid, "/usr/bin/sudo -iu %sadm HDB version", keep_rc=True)
```
This `sap_hdb_version` only requires the list of `SID`.
```python
sapcontrol_getsystemupdatelist = foreach_execute(sap_sid_nr, "/usr/bin/sudo -iu %sadm sapcontrol -nr %s -function GetSystemUpdateList", keep_rc=True)
```
This `sapcontrol_getsystemupdatelist` requires a list of the tuple of `SID` and `SystemNumber`.

And soon, I will add another `foreach_collect` as
```
sap_dev_disp = foreach_collect(sap_sid_sn, '/usr/sap/%s/%s/work/dev_disp', keep_rc=True)
```
which will depend on `SID` and `InstanceName`

I'm not sure how do you mean "just use `SAPHostCtrlInstances` parser", but I think if doing that, I need to re-implement the 3 `foreach_*` specs with 3 `datasource` and in which there must be some duplicated code. Please correct me if I'm wrong. Thanks.

File: insights/specs/default.py [hunk 1]
```diff
@@ -858,10 +859,9 @@ def rhev_data_center(broker):
     rpm_V_packages = simple_command("/bin/rpm -V coreutils procps procps-ng shadow-utils passwd sudo", keep_rc=True)
     rsyslog_conf = simple_file("/etc/rsyslog.conf")
     samba = simple_file("/etc/samba/smb.conf")
-    saphostctrl_listinstances = simple_command("/usr/sap/hostctrl/exe/saphostctrl -function ListInstances")
 
-    @datasource(saphostctrl_listinstances, hostname)
-    def sap_sid_nr(broker):
+    @datasource(SAPHostCtrlInstances, hostname)
+    def sap_instance_info(broker):
```
[the remark this answers is not in the document]
Comment (xiangce) [reply insights/specs/default.py:864]: @csams - Makes sense. I will try to update the parser to provide a friendly API to these `foreach`s. Thanks for the suggestion.

**Comment by xiangce** [comment]
@bfahr , @csams - I changed to use the Sap combiner in the specs, please have a check again. Thanks.

**Comment by bfahr** [comment]
@csams I think the changes look good.

**Pull request merged**